.changelog/unreleased/features/2465-x-vault-update-v1-0-4.md

```diff
@@ -0,0 +1 @@
+* Upgrade `x/vault` module to `v1.0.4` [PR 2465](https://github.com/provenance-io/provenance/pull/2465).
```

Comment:
There's a script that will generate the dependency bump changelog entry.

```console
.changelog/get-dep-changes.sh --pr 2465 --id x-vault-update-v1-0-4
```

Then you can get rid of this entry.